src/api/app/controllers/webui/request_controller.rb

```diff
@@ -521,4 +495,61 @@ def staging_status(request, target_project)
       target_project_staging_url: staging_workflow_path(request.target_project_name)
     }
   end
+
+  def prepare_request_data
```

Comment:
@rubhanazeem Do you plan to break this method up? So that only the data that is required for the individual views is prepared

Reply:
There is a plan to break this method into smaller methods depending on the active tab. Will do it in a following PR